vignettes/download_mail.Rmd

```diff
@@ -0,0 +1,252 @@
+      mbox_file_path: ../../rawdata/helix/pipermail/save_mbox_mail/kaiaulu_202407.mbox
+  # for mod mbox
+  mod_mbox:
+    project_key_1:
+      mailing_list: https://lists.apache.org/list.html?announce@apache.org
+      save_folder_path: ../../rawdata/helix/mod_mbox/save_mbox_mail
+      # mbox_file_path is for use only with parse_mbox() function. It is the file to parse
+      mbox_file_path: ../../rawdata/helix/mod_mbox/save_mbox_mail/kaiaulu_202407.mbox
+```
+
+The most time intensive step you will be required is to locate the URL of the mailing list archive you wish for in the project website. This is specified under `mailing_list`. Note for pipermail this URL should point to the page containing links to the monthly archives (e.g. https://mta.openssl.org/pipermail/openssl-users/), not the top-level mailing list page that contains all the different types of archives (e.g. https://mta.openssl.org/mailman/listinfo/).
+
+
+Note: It is important that the paths specified in save_folder_path and mbox_file_path are accurate and do not conflict between projects.
+
+By organizing the configuration in this way, you can manage multiple projects and mailing lists easily. The notebook reads these parameters and uses them to download and process the archives.
+
+Regardless of which mail archive you choose, the downloaders will store the mail data in monthly files, in a `.mbox` format. This is a simple text file that contains some markings to identify the header of the e-mail containing title, authors, etc. You can open any of the .mbox downloaded files with any text editor. 
+
+## Tools Configuration
+
+In addition to the mailing list configurations, you need to specify the path to the [Perceval](https://github.com/chaoss/grimoirelab-perceval) binary in tools.yml. See the wiki for further details on how to setup third party tools. 
+
+Now, you can load the configurations in your R script or notebook using the following code:
+
+```{r eval=FALSE}
+# Load tools configuration
+tools <- parse_config("../tools.yml")
+parse_perceval_path <- get_tool_project("perceval", tools)
+
+# Load project configuration
+conf <- parse_config("../conf/helix.yml")
+mbox_file_path <- get_mbox_input_file(conf, "project_key_1")
+```
+
+# Downloaders and Refreshers
+
+## Downloaders
+
+With the configurations loaded, we can proceed to download the mailing list archives. The downloaders are responsible for fetching the archives from the specified mailing lists and saving them locally in .mbox format.
+
+### Pipermail Downloader
+
+For Pipermail, we need to specify the project key, which is used to retrieve the configuration parameters for the specific project. The project key is used to identify the project in the configuration file.
+
+Now, we can use the getter functions to retrieve the configuration parameters for the specified project key.
+
+```{r eval=FALSE}
+conf <- parse_config("../conf/helix.yml")
```

Comment:
This is incorrect. The config pipermail points to openssl. You can't use openssl config as helix to run the notebook. Use openssl config instead.. @daomcgill This needs to be properly executed and inspected.